### PR TITLE
Store languages that user understands

### PR DESCRIPTION
**Pull request by lightalloy**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [x] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description
Store languages, that user understands, in the database

Discussion on how to store the languages can be found in[ the issue comments](https://github.com/forem/forem/issues/19846#issuecomment-1700587945)

The pr doesn't include the "default forem language" part (will be implemented separately) and populating users languages table.

## Related Tickets & Documents
- Related Issue #19846

## QA Instructions, Screenshots, Recordings
The pr only provides structure for future usage, so doesn't affect the functionality

## Added/updated tests?
- [x] Yes

### CODE REVIEW

File: app/services/languages/detection.rb [hunk 1]
```diff
@@ -4,6 +4,10 @@ class Detection
 
     PROBABILITY_THRESHOLD = 0.5
 
+    def self.codes
```
Comment (lightalloy) [inline app/services/languages/detection.rb:7]: Can be extracted into a separate class if we need more functionality related to the language codes.

File: db/migrate/20230906112602_create_user_languages.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+class CreateUserLanguages < ActiveRecord::Migration[7.0]
+  def change
+    create_table :user_languages do |t|
+      t.references :user, foreign_key: true, null: false
+      t.string :language
```
Comment (klardotsh) [inline db/migrate/20230906112602_create_user_languages.rb:5]: I suspect this is auto-generated by ActiveRecord this way, but to my eye this seems like a field that should be `null: false` and enforced by Postgres?

File: db/schema.rb [hunk 3]
```diff
@@ -1170,6 +1170,14 @@
     t.index ["blocked_id", "blocker_id"], name: "index_user_blocks_on_blocked_id_and_blocker_id", unique: true
   end
 
+  create_table "user_languages", force: :cascade do |t|
+    t.datetime "created_at", null: false
+    t.string "language"
```
Comment (klardotsh) [inline db/schema.rb:1175]: ditto

**Review by klardotsh — APPROVED** [review]

File: db/migrate/20230906112602_create_user_languages.rb (hunk 2 above)
Comment (lightalloy) [reply db/migrate/20230906112602_create_user_languages.rb:5]: I had a feeling that I forgot something.

**Review by Ridhwana — APPROVED** [review]
Something is up with the Cypress test in the CI, but the code looks good to me.

**Review by jaw6 — APPROVED** [review]

**Pull request merged**